src/appshell/view/appmenumodel.cpp

```diff
@@ -483,8 +486,10 @@ MenuItemList AppMenuModel::workspacesItems() const
         return workspace1->name() < workspace2->name();
     });
 
+    int index = 0;
     for (const IWorkspacePtr& workspace : workspaces) {
         MenuItem item = uiactionsRegister()->action("select-workspace");
+        item.id = QString::fromStdString(item.code) + QString::number(index);
```

Comment:
index always will be 0

Better to move it inside for statement:

for (int i = 0; i < m_workspaces.size(); ++i) {
IWorkspacePtr& workspace = workspaces[i];
item.id = makeId(item.code, i);
...
}